src/groundlight/client.py

```diff
@@ -83,16 +83,20 @@ def _fixup_image_query(self, iq: ImageQuery) -> ImageQuery:  # pylint: disable=n
         return iq
 
     def get_detector(self, id: Union[str, Detector]) -> Detector:  # pylint: disable=redefined-builtin
+        """Get a detector by id"""
```

Comment:
Side question - why is this a Union type to begin with? When would we want to get a detector by detector? Seems like someone copy-pasted that from submit_image_query or something like that.

Reply:
I would think so, except there's the explicit short circuit logic below so this had a bit more thought than just a copy paste